src/game/object/creature.h

```diff
@@ -90,6 +98,31 @@ class Creature : public SpatialObject {
 
     // END Pathfinding
 
+    CombatState cbt_s = CombatState::Idle;
+
+    bool melee_command = false;
+    bool ranged_command = false;
+
+    /* combat animation interruption */
+    bool isInterrupted() {
+        return cbt_s != CombatState::Idle;
+    }
+
+    void setStaggered() {
+        staggered = true;
+    }
+
+    bool isStaggered() {
```

Comment:
Getter functions (`get*`, `is*`, `has*`) should not have side effects. I think, toggle logic should be moved to the calling code.